daml-model/daml/Tests/Onboarding.daml

```diff
@@ -53,62 +53,69 @@ data GuiBackendConfigEntry = GuiBackendConfigEntry with
     name : Text
   deriving (Eq, Show)
 
-
 onboarding : [PartyInfo] -> Script Output
 onboarding partyInfo = do
   scheduler <- allocatePartyOn "Scheduler" (ParticipantName "schedulerParticipant")
   assembler <- allocatePartyOn "Assembler" (ParticipantName "assemblerParticipant")
-
-  (_, entries) <- foldlA
-    (\(parties, entries) info -> do
-      account <- setupAccount scheduler assembler parties info
-      let updatedParties = TextMap.insert info.name account.damlPartyId parties
-      pure (updatedParties, account :: entries))
-    (TextMap.empty, [])
-    partyInfo
-
-  pure $ Output { guiBackendConfig = entries, parties = fmap (.damlPartyId) entries }
-
+  parties <- setupParties partyInfo
+  entries <- mapA (\info -> setupAccount scheduler assembler parties info) partyInfo
```

Comment:
You don't really need `\info -> setupAccount scheduler assembler parties info` but the curried version `setupAccount scheduler assembler parties`).

Reply:
Neat!